lib/samples/add_raster_from_file/add_raster_from_file.dart

```diff
@@ -0,0 +1,98 @@
+// Copyright 2025 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import 'dart:io';
+
+import 'package:arcgis_maps/arcgis_maps.dart';
+import 'package:arcgis_maps_sdk_flutter_samples/common/common.dart';
+import 'package:arcgis_maps_sdk_flutter_samples/utils/sample_data.dart';
+import 'package:arcgis_maps_sdk_flutter_samples/utils/sample_state_support.dart';
+import 'package:flutter/material.dart';
+import 'package:path_provider/path_provider.dart';
+
+class AddRasterFromFile extends StatefulWidget {
+  const AddRasterFromFile({super.key});
+
+  @override
+  State<AddRasterFromFile> createState() => _AddRasterFromFileState();
+}
+
+class _AddRasterFromFileState extends State<AddRasterFromFile>
+    with SampleStateSupport {
+  // Create a controller for the map view.
+  final _mapViewController = ArcGISMapView.createController();
+
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Stack(
+        children: [
+          ArcGISMapView(
+            controllerProvider: () => _mapViewController,
+            onMapViewReady: onMapViewReady,
+          ),
+          // Display a progress indicator and prevent interaction until state is ready.
+          LoadingIndicator(visible: !_ready),
+        ],
+      ),
+    );
+  }
+
+  Future<void> onMapViewReady() async {
+    // Create a map with the ArcGIS ImageryStandard basemap style and set to the map view.
+    final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISImageryStandard);
+
+    // Load the raster layer.
+    final rasterLayer = await loadRasterLayer();
```

Comment:
```suggestion
    final rasterLayer = await loadRasterFromFile();
```
Please could we amend this method name. We aren't actually loading the `RasterLayer` in that method, and we can also emphasize that this method is also doing the file part. It loads the Raster, but just returns the RasterLayer, which gets implicitly loaded when it's added to the map. That's fine as a workflow, I just want to avoid any potential confusion.

Reply:
Done. I totally agree with the assessment and will keep that in mind moving forward.